src/lib/salad/rtree.h

```diff
@@ -284,6 +284,20 @@ rtree_insert(struct rtree *tree, struct rtree_rect *rect, record_t obj);
 bool
 rtree_remove(struct rtree *tree, const struct rtree_rect *rect, record_t obj);
 
+/**
+ * @brief Insert, replace or delete a record in the tree
+ * @return 0 on success, -1 on memory error
+ * @param tree - pointer to a tree
+ * @param old_rect - rectangle of the record to delete
+ * @param old_obj - record to delete
+ * @param new_rect - rectangle to insert
+ * @param new_obj - record to insert
+ * @param removed - set to true if the old_obj is removed
+ */
+int
+rtree_replace(struct rtree *tree, struct rtree_rect *old_rect, record_t old_obj,
```

Comment:
Please re-split. The first commit should introduce this method to the salad library and add a unit test. The second commit should patch the memtx engine to use this method instead of `rtree_insert` + `rtree_delete` + reservation.

Reply:
[Deferred](https://github.com/tarantool/tarantool/pull/10734#discussion_r1847764459) this part of the patch.